### PR TITLE
Tech Docs Review of BESS-KGE docs

### PR DESCRIPTION
**Pull request by jayniep-gc**
This PR contains fixes to the documentation and the Python docstrings. There are also some typo fixes to the Jupyter notebooks.

### CODE REVIEW

**Comment by jayniep-gc** [comment]
Hi Alberto, I've done the review. It went fairly quickly.

Just a note about why I changed the docstrings for any boolean parameters. We follow the [Google Developer Documentation Style Guide](https://developers.google.com/style/api-reference-comments#parameters) and that recommends specifying what the parameter does when it's True and False. So, can you please check that I've got the logic right.

**Comment by AlCatt91** [comment]
Hi Jaynie, thank you for the review! I am going through the changes, I just noticed that there are a few vscode merge editor artifacts left in the README (duplicate parts within <<<<<<< HEAD ... >>>>>>> main ...)

File: besskge/sharding.py [hunk 1]
```diff
@@ -48,14 +48,14 @@ class Sharding:
...
-        Number of entities in the KG.
+        Number of entities in the knowledge graph.
         """
         return len(self.entity_to_shard)
 
     @property
     def max_entity_per_shard(self) -> int:
         """
-        Number of entities (including possibily one final padding) in each shard.
+        Number of entities (including possibly one final padding) in each shard.
```
Comment (AlCatt91) [inline besskge/sharding.py:58]: I realized there was a small mistake in the implementation of the padding, [I pushed a fix for this](https://github.com/graphcore-research/bess-kge/pull/17) - as a result this docstring was modified, so you might need to rebase (sorry!)

File: besskge/sharding.py [hunk 2]
```diff
@@ -341,14 +344,15 @@ def create_from_queries(
...
             "hr" for (h,r,?) queries, "rt" for (?,r,t) queries.
         :param ground_truth: shape: (n_query,)
-            If know, the global ID of the ground truth tail/head.
+            If known, the global ID of the ground truth tail/head.
         :param negative: shape: (N, n_negative)
             Global IDs of negative entities to score against each query,
-            query-specific (N=n_query) or same for all queries (N=1).
-            Defaults to None (i.e. score queries against all entities in the graph).
+            query-specific (N=n_query) or the same for all queries (N=1).
+            Default: None (namely the score queries against all entities in the
```
Comment (AlCatt91) [inline besskge/sharding.py:351]: The additional "the" here doesn't sound quite right to me, I was using "score" as a verb (with "queries" as direct object). Maybe we can rephrase "(namely score each query against all entities...)"

File: besskge/sharding.py [hunk 3]
```diff
@@ -341,14 +344,15 @@ def create_from_queries(
...
+            graph).
         :param negative_type:
-            Score queries only against entities of a specific type.
-            Defaults to None (i.e. score queries against entities of any type).
+            Score queries only against entities of a specific type. Default:
+            None (namely the score queries against entities of any type).
```
Comment (AlCatt91) [inline besskge/sharding.py:355]: Same as above.

**Review by AlCatt91 — COMMENTED** [review]
I double checked everything except for the README (see previous comment), it looks great, thanks! I left just a few minor comments.

File: besskge/sharding.py (hunk 2 above)
Comment (jayniep-gc) [reply besskge/sharding.py:354]: This is my proposal:
```
        :param negative: shape: (N, n_negative)
            Global IDs of negative entities to score against each query. This
            can be query-specific (N=n_query) or the same for all queries (N=1).
            Default: None (namely score each query against all entities in the
            graph).
        :param negative_type:
            Score each query only against entities of a specific type. Default:
            None (namely score each query against entities of any type).
```

**Comment by jayniep-gc** [comment]
Hi Alberto, I figured that you were on holiday 🙂! I've made the changes.

File: README.md [hunk 4]
```diff
@@ -184,17 +186,15 @@ import besskge
...
+1. [KGE training and inference on the OGBL-BioKG dataset](notebooks/1_biokg_training_inference.ipynb) [![Run on Gradient](https://assets.paperspace.io/img/gradient-badge.svg)](https://console.paperspace.com/github/graphcore-research/bess-kge?container=graphcore%2Fpytorch-jupyter%3A3.2.0-ubuntu-20.04&machine=Free-IPU-POD4&file=%2Fnotebooks%2F1_biokg_training_inference.ipynb)
+2. [Link prediction on the YAGO3-10 dataset](notebooks/2_yago_topk_prediction.ipynb) [![Run on Gradient](https://assets.paperspace.io/img/gradient-badge.svg)](https://console.paperspace.com/github/graphcore-research/bess-kge?container=graphcore%2Fpytorch-jupyter%3A3.2.0-ubuntu-20.04&machine=Free-IPU-POD4&file=%2Fnotebooks%2F2_yago_topk_prediction.ipynb)
+3. [FP16 weights and compute on the OGBL-WikiKG2 dataset](notebooks/3_wikikg2_fp16.ipynb) [![Run on Gradient](https://assets.paperspace.io/img/gradient-badge.svg)](https://console.paperspace.com/github/graphcore-research/bess-kge?container=graphcore%2Fpytorch-jupyter%3A3.2.0-ubuntu-20.04&machine=Free-IPU-POD4&file=%2Fnotebooks%2F3_wikikg2_fp16.ipynb)
+
```
[the remark this answers is not in the document]
Comment (AlCatt91) [reply README.md:193]: That's good to know, thanks! I will probably do the same, then.

**Review by AlCatt91 — APPROVED** [review]
Everything looks good to me now, thanks for the review!

**Pull request merged**